### PR TITLE
Move manifest endpoints to their own, independent API group and fix access policies accordingly

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description

We currently have issues with non-admins not being able to access package manifests from the server. This is because the "all manifests" endpoint is part of the _Packages_ API group. In this group, all API controllers inherit from the `PackageControllerBase`, which requires access to the packages section to invoke.

Since base class attributes (`[Authorize]` in this case) cannot be overwritten by specialised classes, we have two choices:

1. Remove the access policy specification from `PackageControllerBase` and decorate all other controllers in the _Packages_ API group instead.
2. Create a new API group for manifests.

Although manifests and packages have a certain cohesion, the manifest endpoints are actually utilised completely independently of the package endpoints. Therefore, option 2 has been chosen.

### Testing this PR

First and foremost, verify that the manifest endpoints are now located in their own _Manifest_ group in the Swagger docs:

image

Now test the access policies:

- `/umbraco/management/api/v1/manifest/manifest` should be able to invoke with _any_ backoffice user (i.e. an editor user)
- `/umbraco/management/api/v1/manifest/manifest/public` should be able to invoke as anonymous.

### CODE REVIEW

File: src/Umbraco.Cms.Api.Management/Controllers/Manifest/PublicManifestManifestController.cs [hunk 1]
```diff
@@ -24,10 +24,10 @@ public PublicPackageManifestController(IPackageManifestService packageManifestSe
     // NOTE: this endpoint is deliberately created as non-paginated to ensure the fastest possible client initialization
     [HttpGet("manifest/public")]
     [MapToApiVersion("1.0")]
-    [ProducesResponseType(typeof(IEnumerable<PackageManifestResponseModel>), StatusCodes.Status200OK)]
-    public async Task<IActionResult> PublicPackageManifests()
+    [ProducesResponseType(typeof(IEnumerable<ManifestResponseModel>), StatusCodes.Status200OK)]
+    public async Task<IActionResult> PublicManifests()
     {
         PackageManifest[] packageManifests = (await _packageManifestService.GetPublicPackageManifestsAsync()).ToArray();
```
Comment (Nuklon) [inline src/Umbraco.Cms.Api.Management/Controllers/Manifest/PublicManifestManifestController.cs:30]: Same here.

File: src/Umbraco.Cms.Api.Management/Controllers/Manifest/PublicManifestManifestController.cs [hunk 2]
```diff
@@ -24,10 +24,10 @@ public PublicPackageManifestController(IPackageManifestService packageManifestSe
...
-        return Ok(_umbracoMapper.MapEnumerable<PackageManifest, PackageManifestResponseModel>(packageManifests));
+        return Ok(_umbracoMapper.MapEnumerable<PackageManifest, ManifestResponseModel>(packageManifests));
```
Comment (Nuklon) [inline src/Umbraco.Cms.Api.Management/Controllers/Manifest/PublicManifestManifestController.cs:31]: Same here.

File: src/Umbraco.Infrastructure/Manifest/PackageManifestService.cs [hunk 3]
```diff
@@ -40,6 +40,9 @@ public async Task<IEnumerable<PackageManifest>> GetAllPackageManifestsAsync()
     public async Task<IEnumerable<PackageManifest>> GetPublicPackageManifestsAsync()
         => (await GetAllPackageManifestsAsync()).Where(manifest => manifest.AllowPublicAccess).ToArray();
 
+    public async Task<IEnumerable<PackageManifest>> GetPrivatePackageManifestsAsync()
+        => (await GetAllPackageManifestsAsync()).Where(manifest => manifest.AllowPublicAccess == false).ToArray();
+
```
Comment (Nuklon) [inline src/Umbraco.Infrastructure/Manifest/PackageManifestService.cs:45]: Given the result type here (same applies above), ToArray can be avoided.

**Comment by kjac** [comment]
Thanks @Nuklon 👍

**Pull request merged**